USING_WITH_PYRIGHT.md

```diff
@@ -0,0 +1,305 @@
+# Pyright and Pylance Together
+
+[Pyright](https://github.com/Microsoft/pyright) is an open source version of a Python type checker that forms the basis for Pylance. Both Pyright and Pylance can be used inside of an IDE to provide a [language server](https://code.visualstudio.com/api/language-extensions/language-server-extension-guide#why-language-server) for Python code.
+
+Pyright can also be used on the [command line](https://github.com/microsoft/pyright/blob/main/docs/command-line.md) to produce json or text output. This can be used in a variety of situations, like a pre-commit hook, or a [github action](https://github.com/jakebailey/pyright-action).
+
+However, Pylance and Pyright don't always produce the same results. The reason for this is a difference in configuration. 
+
+## Stub differences
+
+Pylance comes bundled with a bunch of stubs, usually found here in the installation:
+
+```
+<pylance-extension-install>/dist/bundled
+<pylance-extension-install>/dist/native-stubs
+```
+
+In order for Pyright to produce the same errors as Pylance it need to use the same stub files. 
+
+This might be done by copying the contents of each of those folders to a new folder. 
+
+You'd then reference that new folder in your [pyrightconfig.json](https://microsoft.github.io/pyright/#/configuration)
+
+```json
+{
+    "stubPath": "./pylanceStubs"
+}
+```
+
+Where `pylanceStubs` contains the contents of the `bundled` and `native-stubs` folders.
+
+We have plans to make this easier in the [future](https://github.com/microsoft/pylance-release/discussions/3638).
+
+## Settings differences
+
+Both Pylance and Pyright have [settings](https://github.com/microsoft/pylance-release#settings-and-customization) to control behavior. For the most part, they use the same values, but the few differences do have an impact on the analysis.

```

Comment:
It might be worth talking about "language server settings" versus "pyright configuration settings". The latter are found in a `pyrightconfig.json` or `pyproject.toml` file and tend to override the former, although there are cases where the two are composed together. (It's complicated.) In general, if users want consistent behavior between pyright and pylance, I recommend that they do not use language server (e.g. VS Code) settings and instead set all of their configuration settings in the `pyrightconfig.json` or `pyproject.toml` file.

If users follow this recommendation, then you can remove most of the text below because pyright and pylance will both pick up these settings and act consistently between the two.

Reply:
Maybe I should add another entry to the table to specify the 'pyrightconfig.json' setting

Reply:
I don't think people using the IDE will want a pyrightconfig.json that can also be impacted by settings.json? Not sure. Does pyrightconfig.json work in the .vscode folder or does it have to be in the root of the project?

Reply:
The `pyrightconfig.json` or `pyproject.toml` must be in the root of the project. This is consistent with other tools (e.g. `tsconfig.json`). This file is meant to be checked in and shared across all project contributors, whereas the ".vscode/settings.json" file is often not checked in (since it contains per-user settings in many cases).

We should confirm my assumption that "there are cases where the two are composed together". Perhaps that's not the case and `pyrightconfig.json` and `pyproject.toml` configuration settings always prevail. I think there are cases where that's not true though. For example, if you specify a language server setting but don't specify the corresponding setting in the config file, I think the language server setting prevails (as opposed to the default value for that setting).

Reply:
In any case, what I say above is the advice I recommend to people who want the same behavior from pyright and pylance: don't use language server settings.

Reply:
I tried the following:

- pyrightconfig.json = reportUndefinedVariables = none
- user settings.json = reportUndefinedVariables = error
- workspace settings.json = reportUndefinedVariables = error

pyrightconfig.json wins.
 
Recommending using pyrighconfig.json seems like a good idea because it will always trump everything else.